exe-nix-buildkite/Main.hs

```diff
@@ -62,6 +63,40 @@ withTime label k = do
   where
     click = getTime Monotonic
 
+nixInstantiate :: String -> IO [String]
+nixInstantiate jobsExpr = withTime "nix-instantiate" (Prelude.lines <$> readProcess "nix-instantiate" [ jobsExpr ] "")
+
+nixBuildDryRun :: String -> IO [String]
+nixBuildDryRun jobsExpr = withTime "nix-build --dry-run" do
+  (_stdin, _stdout, stderrHndl, _prchndl) <- createProcess $ (proc "nix-build" ["--dry-run", jobsExpr]) { std_err = CreatePipe }
+
+  inputLines <- Prelude.lines <$> case stderrHndl of
+    Just hndl -> hGetContents hndl
+    Nothing -> pure []
+  -- See Note: [nix-build --dry-run output]
+  let stripLeadingWhitespace = dropWhile (==' ')
+  let theseLine = List.isPrefixOf "these"
+
+  -- dump the output to stderr
+  mapM_ (hPutStrLn stderr) inputLines
```

Comment:
Is this something we want to keep, or just here for debug? I don't think we have the same for `nix-instantiate`

Reply:
Ah I see. I mostly did it because we implicitly had this before with instantiate, and because I like to see these traces in the logs. When it takes a bit longer, it can give a hint as to what's causing it

Reply:
Hmm, is that true? I don't see any output in https://buildkite.com/organizations/circuithub/pipelines/circuithub/builds/42646/jobs/01957fed-7f71-43e1-a1f6-9b86189137c9/log, which I believe is still using `nix-instantiate`. Anyway, I don't feel too strongly here, but could maybe do with a bit more explanation in a comment.

Reply:
ah!